service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -28,69 +35,198 @@
-                observationMapper.mapObservationToCompoundStatement(
-                    observationAssociatedWithSpecimen,
-                    observations
-                )
-            );
-        });
-
-        final IdMapper idMapper = messageContext.getIdMapper();
+        return observationsAssociatedWithSpecimen.stream().map(
+            observationAssociatedWithSpecimen -> observationMapper.mapObservationToCompoundStatement(
+                observationAssociatedWithSpecimen,
+                observations
+            ))
+            .collect(Collectors.joining());
+    }
 
-        var specimenCompoundStatementTemplateParameters = SpecimenCompoundStatementTemplateParameters.builder()
-            .compoundStatementId(idMapper.getOrNew(ResourceType.Specimen, specimen.getId()))
-            .diagnosticReportIssuedDate(DateFormatUtil.toHl7Format(diagnosticReport.getIssuedElement()))
-            .accessionIdentifier(specimen.getAccessionIdentifier().getValue())
-            .effectiveTime(prepareEffectiveTimeElement(specimen))
-            .observations(observationsBlock.toString());
-
-        if (specimen.getType().hasCoding()) {
-            specimen.getType().getCoding().stream()
-                .findFirst()
-                .ifPresent(coding ->
-                    specimenCompoundStatementTemplateParameters.type(
-                        coding.getDisplay()
-                    )
-            );
-        } else if (specimen.getType().hasText())  {
-            specimenCompoundStatementTemplateParameters.type(specimen.getType().getText());
+    private Optional<String> buildPertinentInformation(Specimen specimen) {
+        PertinentInformationBuilder pertinentInformationBuilder = new PertinentInformationBuilder();
+        if (specimen.hasCollection()) {
+            Specimen.SpecimenCollectionComponent collection = specimen.getCollection();
+
+            collection.getExtensionsByUrl(FASTING_STATUS_URL).stream().findFirst()
+                .ifPresent(extension -> {
+                    Type value = extension.getValue();
+                    if (value instanceof CodeableConcept) {
+                        pertinentInformationBuilder.fastingStatus((CodeableConcept) value);
+                    } else if (value instanceof Duration) {
+                        pertinentInformationBuilder.fastingDuration((Duration) value);
+                    }
+                });
+
+            if (collection.hasQuantity()) {
+                pertinentInformationBuilder.quantity(collection.getQuantity());
+            }
+
+            if (collection.hasBodySite()) {
+                pertinentInformationBuilder.collectionSite(collection.getBodySite());
+            }
         }
 
-        return TemplateUtils.fillTemplate(
-            SPECIMEN_COMPOUND_STATEMENT_TEMPLATE,
-            specimenCompoundStatementTemplateParameters.build()
-        );
+        specimen.getNote().forEach(note -> pertinentInformationBuilder.note(note.getText()));
```

Comment:
Perhaps a bit more idiomatic:
```suggestion
        specimen.getNote().stream()
            .map(Annotation::getText)
            .forEach(pertinentInformationBuilder::note);
```